tests/test_service_resources.py

```diff
@@ -40,7 +40,12 @@ def test_error_result_reports_error_on_success(self):
 
     def test_invalid_json_reports_error_on_success(self):
         self.mock_ok_service(data=None)
+        json_decoding_error_messages = {
+            'python2_error': 'No JSON object could be decoded',
+            'python3_error': 'Expecting value: line 1 column 1 (char 0)',
+        }
+
         result = self.service.check()
         self.assertEqual(result['status'], ERROR)
         self.assertEqual(result['result'], None)
-        self.assertEqual(result['error'], 'No JSON object could be decoded')
+        self.assertIn(result['error'], json_decoding_error_messages.values())
```

Comment:
I've recently been sold on the idea that tests should [expect literals](http://blog.jayfields.com/2008/02/testing-expect-literals.html). If we handle the decode error in the service canary as a special case, we can guarantee a consistent error message across Python versions, right?